client/blocks/plan-thank-you-card/index.jsx

```diff
@@ -0,0 +1,123 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSite } from 'state/ui/selectors';
+import Gridicon from 'components/gridicon';
+import { getCurrentPlan } from 'state/sites/plans/selectors';
+import { fetchSitePlans } from 'state/sites/plans/actions';
+import { getPlanBySlug, getPlans } from 'state/plans/selectors';
+import { requestPlans } from 'state/plans/actions';
+
+class PlanThankYouCard extends Component {
+	static propTypes = {
+		selectedSite: PropTypes.object
+	};
+
+	componentDidMount() {
+		if ( ! this.props.plan ) {
+			this.props.fetchSitePlans( this.props.selectedSite );
+		}
+
+		if ( ! this.props.plansFetched ) {
+			this.props.requestPlans();
+		}
+	}
+
+	render() {
+		// Non standard gridicon sizes are used here because we use them as background pattern with various sizes and rotation
+		/* eslint-disable wpcalypso/jsx-gridicon-size */
+		return (
+			<div className="plan-thank-you-card">
+				<div className="plan-thank-you-card__header">
+					<Gridicon className="plan-thank-you-card__main-icon" icon="checkmark-circle" size={ 140 } />
+					<div className="plan-thank-you-card__plan-name">
+
+					</div>
+					{ ( ! this.props.planDetails ) ? (
+						<div className="plan-thank-you-card__plan-name is-placeholder"></div>
```

Comment:
I would do the placeholder logic a bit differently. In `mapStateToProps` (the first fn passed to `connect`), I would return a prop called `isPlaceholder` based on whether `planDetails` and/or `plan` is `null/falsey`. I think it's more readable and other programmers can see from a quick look why is there the condition:

```js
{ isPlaceholder
   ? <div className="...__placeholder"></div>
   : <not placeholder>
}
```